test/with_dummyserver/test_socketlevel.py

```diff
@@ -1428,12 +1428,12 @@ def socket_handler(listener):
 
 class TestBrokenHeaders(SocketDummyServerTestCase):
 
-    def _test_broken_header_parsing(self, headers):
+    def _test_broken_header_parsing(self, headers, unparsed_data_check=None):
         self.start_response_handler((
            b'HTTP/1.1 200 OK\r\n'
            b'Content-Length: 0\r\n'
            b'Content-type: text/plain\r\n'
-           ) + b''.join(headers) + b'\r\n'
+           ) + b''.join(headers) + b'\r\n\r\n'
```

Comment:
Ok yeah if you look at https://github.com/python-trio/urllib3/commit/285889d57f8bb00116617dc654124a4c7d223fed then it has `b'\r\n'` here. I'm not sure how or why we diverged, but the changes below are assuming it's `b'\r\n'`

Reply:
For reference, I think you meant https://github.com/python-trio/urllib3/pull/95/commits/f4efcca559cc753f878e0a73e28dd6e795d0d052